imap/browse.c

```diff
@@ -220,6 +220,21 @@ int imap_browse(const char *path, struct BrowserState *state)
   if (!adata)
     goto fail;
 
+  if (C_ImapListSubscribed)
+  {
+    const char *lsub_cmd = "LSUB";
+
+    /* RFC3348 section 3 states LSUB is unreliable for hierarchy information.
+     * The newer LIST extensions are designed for this.  */
+    if (adata->capabilities & IMAP_CAP_LIST_EXTENDED)
+      lsub_cmd = "LIST (SUBSCRIBED)";
+    mutt_str_strfcpy (list_cmd, lsub_cmd, sizeof (list_cmd));
+  }
+  else
+  {
+    mutt_str_strfcpy (list_cmd, "LIST", sizeof (list_cmd));
+  }
```

Comment:
Cool! Would the code be more readable if we either use `mutt_str_strfcpy` directly in all branches or set the cmd in all branches and call `mutt_str_strfcpy` only once?

version 1:

```c
if (C_ImapListSubscribed)
{
  /* RFC3348 section 3 states LSUB is unreliable for hierarchy information.
   * The newer LIST extensions are designed for this.  */
  if (adata->capabilities & IMAP_CAP_LIST_EXTENDED)
    mutt_str_strfcpy (list_cmd, "LIST (SUBSCRIBED)", sizeof (list_cmd));
  else
    mutt_str_strfcpy (list_cmd, "LSUB", sizeof (list_cmd));
}
else
{
  mutt_str_strfcpy (list_cmd, "LIST", sizeof (list_cmd));
}
```

version 2:
```c
const char *cmd = NULL;
if (C_ImapListSubscribed)
{
  /* RFC3348 section 3 states LSUB is unreliable for hierarchy information.
   * The newer LIST extensions are designed for this.  */
  if (adata->capabilities & IMAP_CAP_LIST_EXTENDED)
    cmd = "LIST (SUBSCRIBED)";
  else
    cmd = "LSUB";
}
else
{
  cmd = "LIST";
}
mutt_str_strfcpy (list_cmd, cmd, sizeof (list_cmd));
```

version 3
```c
mutt_str_strfcpy (
  list_cmd, 
  !C_ImapListSubscribed ? "LIST" : (adata->capabilities & IMAP_CAP_LIST_EXTENDED) ? "LIST (SUBSCRIBED)" : "LSUB"
  sizeof (list_cmd));
```

Hint: one of the versions is a joke.

Reply:
Hmm... 2) _does_ look nicer.